scaletest/dashboard/chromedp.go

```diff
@@ -0,0 +1,216 @@
+package dashboard
+
+import (
+	"context"
+	"math/rand"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/chromedp/cdproto/cdp"
+	"github.com/chromedp/cdproto/network"
+	"github.com/chromedp/chromedp"
+	"golang.org/x/xerrors"
+)
+
+// Action is just a function that does something.
+type Action func(ctx context.Context) error
+
+// Selector locates an element on a page.
+type Selector string
+
+// Target is a thing that can be clicked.
+type Target struct {
+	// Label is a human-readable label for the target.
+	Label Label
+	// ClickOn is the selector that locates the element to be clicked.
+	ClickOn Selector
+	// WaitFor is a selector that is expected to appear after the target is clicked.
+	WaitFor Selector
+}
+
+// Label identifies an action.
+type Label string
+
+var defaultTargets = []Target{
+	{
+		Label:   "workspace_list",
+		ClickOn: `nav a[href="/workspaces"]:not(.active)`,
+		WaitFor: `tr[role="button"][data-testid^="workspace-"]`,
+	},
+	{
+		Label:   "starter_templates",
+		ClickOn: `a[href="/starter-templates"]`,
+		WaitFor: `a[href^="/starter-templates/"]`,
+	},
+	{
+		Label:   "workspace_details",
+		ClickOn: `tr[role="button"][data-testid^="workspace-"]`,
+		WaitFor: `tr[role="button"][data-testid^="build-"]`,
+	},
+	{
+		Label:   "workspace_build_details",
+		ClickOn: `tr[role="button"][data-testid^="build-"]`,
+		WaitFor: `*[aria-label="Build details"]`,
+	},
+	{
+		Label:   "template_list",
+		ClickOn: `nav a[href="/templates"]:not(.active)`,
+		WaitFor: `tr[role="button"][data-testid^="template-"]`,
+	},
+	{
+		Label:   "template_docs",
+		ClickOn: `a[href^="/templates/"][href$="/docs"]:not([aria-current])`,
+		WaitFor: `#readme`,
+	},
+	{
+		Label:   "template_files",
+		ClickOn: `a[href^="/templates/"][href$="/docs"]:not([aria-current])`,
+		WaitFor: `.monaco-editor`,
+	},
+	{
+		Label:   "template_versions",
+		ClickOn: `a[href^="/templates/"][href$="/versions"]:not([aria-current])`,
+		WaitFor: `tr[role="button"][data-testid^="version-"]`,
+	},
+	{
+		Label:   "template_version_details",
+		ClickOn: `tr[role="button"][data-testid^="version-"]`,
+		WaitFor: `.monaco-editor`,
+	},
+	{
+		Label:   "user_list",
+		ClickOn: `nav a[href^="/users"]:not(.active)`,
+		WaitFor: `tr[data-testid^="user-"]`,
+	},
+}
+
+// ClickRandomElement returns an action that will click an element from defaultTargets.
+// If no elements are found, an error is returned.
+// If more than one element is found, one is chosen at random.
```

Comment:
Are we sure we want random here? I think randomness can be good out of a discovering there's a problem-perspective, but if it can produce different kinds of loads during different scale test runs, that might not be ideal for analysis and/or repeating the result.

Reply:
Alternatively, I can make the elements target specific children (e.g. `:nth-child(1)`) or migrate them to XPath instead.

Reply:
Added `--rand-seed` to specify PRNG seed.